operators/src/processing/reprojection.rs

```diff
@@ -191,24 +204,263 @@ where
     }
 }
 
+#[typetag::serde]
+impl RasterOperator for Reprojection {
+    fn initialize(
+        self: Box<Self>,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<InitializedRasterOperator>> {
+        ensure!(
+            self.vector_sources.is_empty(),
+            crate::error::InvalidNumberOfVectorInputs {
+                expected: 0..0,
+                found: self.vector_sources.len()
+            }
+        );
+        ensure!(
+            !self.raster_sources.is_empty(),
+            crate::error::InvalidNumberOfRasterInputs {
+                expected: 1..1,
+                found: self.raster_sources.len()
+            }
+        );
+
+        let initialized_raster_sources = self
+            .raster_sources
+            .into_iter()
+            .map(|o| o.initialize(context))
+            .collect::<Result<Vec<Box<InitializedRasterOperator>>>>()?;
+
+        let in_desc: &RasterResultDescriptor = initialized_raster_sources[0].result_descriptor();
+        let out_desc = RasterResultDescriptor {
+            spatial_reference: self.params.target_spatial_reference.into(),
+            data_type: in_desc.data_type,
+            measurement: in_desc.measurement.clone(),
+        };
+
+        let state = RasterReprojectionState {
+            source_srs: Option::from(in_desc.spatial_reference).unwrap(),
+            target_srs: self.params.target_spatial_reference,
+            tiling_spec: context.tiling_specification(),
+        };
+
+        Ok(
+            InitializedRasterReprojection::new(out_desc, initialized_raster_sources, vec![], state)
+                .boxed(),
+        )
+    }
+}
+
+impl InitializedOperator<RasterResultDescriptor, TypedRasterQueryProcessor>
+    for InitializedRasterReprojection
+{
+    // i know there is a macro somewhere. we need to re-work this when we have the no-data value anyway.
+    #[allow(clippy::clippy::too_many_lines)]
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let q = self.raster_sources[0].query_processor()?;
+
+        let s = self.state;
+
+        Ok(match self.result_descriptor.data_type {
+            geoengine_datatypes::raster::RasterDataType::U8 => {
+                let qt = q.get_u8().unwrap();
+                TypedRasterQueryProcessor::U8(Box::new(RasterReprojectionProcessor::new(
+                    qt,
+                    s.source_srs,
+                    s.target_srs,
+                    s.tiling_spec,
+                    0_u8,
```

Comment:
Do you always use 0 as NO DATA because you don't know the value?

Reply:
yes, we don't know the NO DATA VALUE here

Reply:
cf. https://github.com/geo-engine/geoengine/pull/186#discussion_r607960145

Reply:
there is a new task to add NO DATA VALUEE to RasterResultDescriptor